Added 'Open External' keyboard shortcut

Krakor92 opened the pull request:
Hello,

Following this issue I recently opened (#495), I added the keyboard shortcut "Open external" to the list of already existing shortcuts.
I set the default key combination to "Ctrl + Enter", I also thought of setting "Ctrl + Space" but after trying it, I found the former more natural.
Also, compared to the default "Right Click -> Open External" on a single entry, this shortcut opens all selected files.

As I did not find a contribution section, neither in the readme nor in the wiki, I do not know if you accept such pr. If not, I'm sorry for the inconvenience.

RvanderLaan commented on src/frontend/stores/UiStore.ts:
```diff
@@ -345,6 +348,16 @@ class UiStore {
+    }
+
+    const absolutePaths = Array.from(this.fileSelection, (file) => file.absolutePath);
+    absolutePaths.forEach((path) => shell.openExternal(`file://${path}`).catch(console.error));
```
as much as I don't like arbitrary hardcoded limits, it might be good to set one here. We don't any anyone's computer to explode when they have hundreds or thousands of images selected. 
Does 25 or so seem reasonable to you?

RvanderLaan reviewed (changes requested):
PRs are most certainly welcome, thanks for taking the time! We should indeed add docs about contributing 

You could also replace the function on the "Open External" menu item (`frontend/containers/ContentView/menu-items.tsx`) to use your function, so the hotkey and menu item behave the same, that would be more natural too

Besides that and one other possible tweak, it's looking good - nice work! 👍

Krakor92 commented:
Instead of limiting to a certain threshold, I think it is better to let the user decide. So I added a little warning dialog when there are at least 10 files selected before doing the `Open external` command.

![Image 5](https://user-images.githubusercontent.com/24924824/182443223-057e4bfc-903c-4f0b-ba71-18a4e991a1d9.png)

However, if you think it is still too dangerous, I can force the opening to stop after the 25th file (I should then add this info to the warning text of the dialog).

- I'm not sure about the title and description of the warning though. Definitly change them if you have better ideas.
- I've fixed the menu-item function, now it calls the UiStore function instead of implementing its own
- (I also just realized that I didn't push on a new branch when making this pr, i'm sorry for that ^^')

---

I thought of moving the `SubLocationExclusion` component from RemovalAlert.tsx to this new WarningAlert.tsx file since its type is "warning" but I don't know if it's a good decision.

I also wanted to give another class to the div with the filenames instead of `deletion-confirmation-list` but I'm not sure if it makes sense in this case

Finally, It's out of the scope of this pr but my last commit fixes a bug i've encountered while trying to delete files. Basicly, I couldn't exit the removal dialog box saying "Are you sure you want to delete" after pressing `Esc` key (which deselects all files)
However, to fix the bug properly: key presses should focus on the visible dialog instead of keeping the focus on the gallery (I don't know how to do that right now).

RvanderLaan commented on src/frontend/components/WarningAlert.tsx:
```diff
@@ -0,0 +1,41 @@
+        uiStore.closeManyExternalFiles();
+      }}
+    >
+      <p>If your hardware is not powerful enough, it could overload your CPU capacity!</p>
```
```suggestion
      <p>This may severely slow down your computer, to the point of it becoming unresponsive.</p>
```

RvanderLaan reviewed (changes requested):
That's even better, I was thinking this approach that after I posted that last suggestion :clap: 

Just left some suggestions for the textual content, the implementation looks good!

That SubLocationExclusion component indeed doesn't belong there with the current naming. They're all basically just confirmation modals, it needs some more thorough refactoring. That also goes for `deletion-confirmation-list`. Better to leave that for later, thanks for pointing it out :+1:  

I'll take a look at that `Esc` listener issue this weekend, we've had some issues with some events being listened globally for various reasons..
